app/javascript/App/bundles/ExchangeListings/client.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component, Fragment } from 'react'
+import debounce from 'debounce'
+import LayoutDesktop from '../../components/LayoutDesktop'
+import LayoutTablet from '../../components/LayoutTablet'
+import LayoutMobile from '../../components/LayoutMobile'
+import ListingsHeader from './components/ListingsHeader'
+import ListingsList from './components/ListingsList'
+import BodySection from './components/BodySection'
+
+class ExchangeListingsPage extends Component {
+  state = { showFilterPanel: false }
+
+  toggleFilterPanel() {
+    this.setState({
+      showFilterPanel: !this.state.showFilterPanel,
+    })
```

Comment:
When state depends on previous state it's highly recommended to use callback b/c by the time next state is being set `this.state` might be not what you expect (`setState` is async):

```js
this.setState(state => ({ showFilterPanel: !state.showFilterPanel }))
```